exp/query/lsm_fetcher.go

```diff
@@ -85,11 +77,11 @@ func NewLSMFetcherWithCache(basePath string, upstream LSMDownloader, cache LSMCa
 // 2. If cache is enabled and not found in cache, fetch from upstream.
 // 3. If cache is enabled and found in the cache but local version is outdated, fetch from the upstream and remove old version.
 // 4. if cache is enabled, found in the cache and local version is up to date, return from the cache.
-func (l *LSMFetcher) Fetch(ctx context.Context, collection, tenant string, version uint64) (*lsmkv.Store, string, error) {
-	// unique key for (collectin, tenant, version). this key make sure only one in-flight requests hitting upstream.
-	key := fmt.Sprintf("%s-%s-%d", collection, tenant, version)
+func (l *LSMFetcher) Fetch(ctx context.Context, node, collection, tenant string, version uint64) (*lsmkv.Store, string, error) {
+	// unique key for (collection, tenant, version). this key make sure only one in-flight requests hitting upstream.
+	key := fmt.Sprintf("%s-%s-%s-%d", node, collection, tenant, version)
```

Comment:
```suggestion
	// unique key for (node, collection, tenant, version). this key make sure only one in-flight requests hitting upstream.
	key := fmt.Sprintf("%s-%s-%s-%d", node, collection, tenant, version)
```
@reyreaud-l do you think including the node in the key is necessary? or should we just assume the node doesn't matter? i think we'll usually get the same node for a given collection/tenant, but i'm trying to think through what happens if that node changes. @kavirajk any thoughts?

Reply:
it totally depends what we need to do if the node changes with same version? we call upstream or no?

I mean given, the replicas can be diverged, still I would love to call upstream only if the version is changed. So basically I would prefer to keep the node out of the key. 

Also curious about the rationale why @reyreaud-l included it.